experimental/LieAlgebras/src/LieAlgebraIdeal.jl

```diff
@@ -0,0 +1,283 @@
+  io = pretty(io)
+  if get(io, :supercompact, false)
+    print(io, LowercaseOff(), "Lie algebra ideal")
+  else
+    print(io, LowercaseOff(), "Lie algebra ideal over ", Lowercase())
+    print(IOContext(io, :supercompact => true), base_lie_algebra(I))
+  end
+end
+
+###############################################################################
+#
+#   Comparisons
+#
+###############################################################################
+
+function Base.issubset(
+  I1::LieAlgebraIdeal{C,LieT}, I2::LieAlgebraIdeal{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+  @req base_lie_algebra(I1) === base_lie_algebra(I2) "Incompatible Lie algebras."
+  return all(in(I2), gens(I1))
+end
+
+function Base.:(==)(
+  I1::LieAlgebraIdeal{C,LieT}, I2::LieAlgebraIdeal{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+  base_lie_algebra(I1) === base_lie_algebra(I2) || return false
+  gens(I1) == gens(I2) && return true
+  return basis_matrix(I1) == basis_matrix(I2)
+end
+
+function Base.hash(I::LieAlgebraIdeal, h::UInt)
+  b = 0xd745d725d0b66431 % UInt
+  h = hash(base_lie_algebra(I), h)
+  h = hash(gens(I), h)
+  h = hash(basis_matrix(I), h)
+  return xor(h, b)
+end
+
+###############################################################################
+#
+#   Ideal membership
+#
+###############################################################################
+
+@doc raw"""
+    in(x::LieAlgebraElem, I::LieAlgebraIdeal) -> Bool
+
+Return `true` if `x` is in the ideal `I`, `false` otherwise.
+"""
+function Base.in(x::LieAlgebraElem, I::LieAlgebraIdeal)
+  return can_solve(basis_matrix(I), _matrix(x); side=:left)
+end
+
+###############################################################################
+#
+#   Constructions
+#
+###############################################################################
+
+function Base.:+(
+  I1::LieAlgebraIdeal{C,LieT}, I2::LieAlgebraIdeal{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+  @req base_lie_algebra(I1) === base_lie_algebra(I2) "Incompatible Lie algebras."
+  return ideal(base_lie_algebra(I1), [gens(I1); gens(I2)])
+end
+
+@doc raw"""
+    bracket(I1::LieAlgebraIdeal, I2::LieAlgebraIdeal) -> LieAlgebraIdeal
+
+Return $[I_1,I_2]$.
+"""
+function bracket(
+  I1::LieAlgebraIdeal{C,LieT}, I2::LieAlgebraIdeal{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+  @req base_lie_algebra(I1) === base_lie_algebra(I2) "Incompatible Lie algebras."
+  return ideal(base_lie_algebra(I1), [x * y for x in gens(I1) for y in gens(I2)])
+end
+
+###############################################################################
+#
+#   Important ideals and subalgebras
+#
+###############################################################################
+
+@doc raw"""
+    normalizer(L::LieAlgebra, I::LieAlgebraIdeal) -> LieSubalgebra
+
+Return the normalizer of `I` in `L`, i.e. $\{x \in L \mid [x, I] \subseteq I\} = L$.
+"""
+function normalizer(L::LieAlgebra, I::LieAlgebraIdeal)
+  return sub(L)
```

Comment:
The code says that the normalizer of an ideal is the full Lie algebra, the documentation states the definition of a normalizer; this looks strange.

It should be checked that `I` is really an ideal in `L`.

Reply:
For subalgebras, the corresponding function actually calculates something. As the normalizer of an ideal always is the full Lie algebra, there is nothing to check here. I adapted the docstring to reflect that in 1e40cb835cd2452e348c553d951d4ca2abd8e7e8

Reply:
Thanks.
It is still possible that someone enters an `I` that is an ideal in some other Lie algebra, not in `L`.

Reply:
Oh yeah, that's true. I will add a check for that after lunch

Reply:
Added in e7f653429c23a0070ea08f84c19924617d85134e